cogs/welcome/welcome.py

```diff
@@ -28,8 +31,23 @@ class Welcome(commands.Cog):  # pylint: disable=too-many-instance-attributes
     def __init__(self, bot: Red):
         self.bot = bot
         self.config = Config.get_conf(self, identifier=5842647, force_registration=True)
+
         self.config.register_guild(**DEFAULT_GUILD)
 
+        self.dataDir = data_manager.cog_data_path(cog_instance=self)
+        self.imgDir = self.dataDir / WELCOME_IMG_FOLDER
+        self.bundled_assets = data_manager.bundled_data_path(self)
+
+        # create folder to hold welcome images
+        try:
+            os.makedirs(self.imgDir, exist_ok=True)
+        except OSError as error:
+            LOGGER.error(
+                "Could not create folder for images!",
+                exc_info=True,
+            )
+            raise OSError("Could not create folder for images!")
```

Comment:
Re-raising an exception should include the original exception as context. Ref: https://docs.python.org/3.9/library/exceptions.html#exception-context

Also I would advise against doing an `OSError` here because its constructor expects some meaningful args. Ref: https://docs.python.org/3.9/library/exceptions.html#OSError

We can do `RuntimeError` here, and extract the error message to a variable for reuse.

```suggestion
        except OSError as error:
            errorMessage = "Could not create folder for images!"
            LOGGER.error(errorMessage, exc_info=True)
            raise RuntimeError(errorMessage) from error
```